src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_TopLevelStatements.cs

```diff
@@ -0,0 +1,180 @@
+            var globalStatements = GetGlobalStatements(typeDeclaration, methodDeclaration);
+
+            var namespaceDeclaration = typeDeclaration.Parent as NamespaceDeclarationSyntax;
+            if (namespaceDeclaration != null &&
+                namespaceDeclaration.Members.Count >= 2)
+            {
+                // Our parent namespace has another symbol in it.  Keep the namespace declaration around, removing only
+                // the existing Program type from it.
+                editor.RemoveNode(typeDeclaration);
+                editor.ReplaceNode(
+                    root,
+                    (current, _) =>
+                    {
+                        var currentRoot = (CompilationUnitSyntax)current;
+                        return currentRoot.WithMembers(currentRoot.Members.InsertRange(0, globalStatements));
+                    });
+            }
+            else if (namespaceDeclaration != null)
+            {
+                // we had a parent namespace, but we were the only thing in it.  We can just remove the namespace entirely.
+                editor.ReplaceNode(
+                    root,
+                    root.ReplaceNode(namespaceDeclaration, globalStatements));
+            }
+            else
+            {
+                // type wasn't in a namespace.  just remove the type and replace it with the new global statements.
+                editor.ReplaceNode(
+                    root, root.ReplaceNode(typeDeclaration, globalStatements));
+            }
+
+            return editor.GetChangedRoot();
+        }
+
+        private static ImmutableArray<GlobalStatementSyntax> GetGlobalStatements(TypeDeclarationSyntax typeDeclaration, MethodDeclarationSyntax methodDeclaration)
+        {
+            using var _ = ArrayBuilder<StatementSyntax>.GetInstance(out var statements);
+            foreach (var member in typeDeclaration.Members)
+            {
+                if (member == methodDeclaration)
+                {
+                    // when we hit the 'Main' method, then actually take all its nested statements and elevate them to
+                    // top-level statements.
+                    Contract.ThrowIfNull(methodDeclaration.Body); // checked by analyzer
+
+                    // move comments on the method to be on it's first statement.
+                    if (methodDeclaration.Body.Statements.Count > 0)
+                        statements.AddRange(methodDeclaration.Body.Statements[0].WithPrependedLeadingTrivia(methodDeclaration.GetLeadingTrivia()));
+
+                    statements.AddRange(methodDeclaration.Body.Statements.Skip(1));
+                    continue;
+                }
+
+                // hit another member, must be a field/method.
+                if (member is FieldDeclarationSyntax fieldDeclaration)
+                {
+                    // Convert fields into local statements
+                    statements.Add(LocalDeclarationStatement(fieldDeclaration.Declaration)
```

Comment:
Should these be inserted into the start of the `statements` collection?

eg we don't want:
```cs
class Program
{
    private static void Main(string[]  args)
    {
        Console.WriteLine(_foo);
    }

    private static string _foo = "Hello World";
}
```

to become:
```cs
Console.WriteLine(_foo);

string _foo = "Hello World";
```

Reply:
yes.  Very good call.  will fixup.